addons/hr/models/hr_department.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models, _
+from openerp.exceptions import ValidationError
+
+
+class HrDepartment(models.Model):
+
+    _name = "hr.department"
+    _description = "HR Department"
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    name = fields.Char(string='Department Name', required=True)
+    complete_name = fields.Char(compute='_compute_complete_name', string='Name')
```

Comment:
use display_name field instead of introduce new complete field for display full name.